### PR TITLE
Add more types to OperatorCompiler

### PR DESCRIPTION
**Pull request by alexfikl**
* Adds more types.
* Changes some `IdentityMapper.__init__` to `super()`.
* Some f-strings here and there.

### CODE REVIEW

File: pytential/symbolic/compiler.py [hunk 1]
```diff
@@ -465,44 +464,46 @@ def execute(self, exec_mapper, pre_assign_check=None):
...
+        return DependencyMapper(
+                # include_operator_bindings=False,
                 include_lookups=False,
                 include_subscripts=include_subscripts,
                 include_calls="descend_args")
 
-        return self.dep_mapper
+    @property
+    def dep_mapper(self):
+        return self.dep_mapper_factory()
```
Comment (alexfikl) [inline pytential/symbolic/compiler.py:499]: This is probably a behavior change, but I didn't really understand what the intention of the original code was.

It gets memoized, but it also sets that `dep_mapper` attribute based on `include_subcripts` (?)

Comment (inducer) [reply pytential/symbolic/compiler.py:499]: I agree that the old code looks might fishy.

**Comment by inducer** [comment]
I was curious what it would take to check these types, which led to #123. (Answer: quite a bit!)

**Comment by inducer** [comment]
Thanks, LGTM!

**Comment by alexfikl** [comment]
> I was curious what it would take to check these types, which led to #123. (Answer: quite a bit!)

Yeah, I imagine it would take a bit of work to get anything useful, since `pymbolic` isn't typed either :(.

**Pull request merged**